evap/staff/tools.py

```diff
@@ -292,3 +301,21 @@ def find_next_unreviewed_evaluation(semester, excluded):
         .filter(contributions__textanswer_set__state=TextAnswer.State.NOT_REVIEWED) \
         .annotate(num_unreviewed_textanswers=Count("contributions__textanswer_set")) \
         .order_by('vote_end_date', '-num_unreviewed_textanswers').first()
+
+
+def remove_user_from_represented_and_ccing_users(user, ignored_users=None, test_run=False):
+    remove_messages = []
+    ignored_users = ignored_users or []
+    for represented_user in user.represented_users.exclude(id__in=[user.id for user in ignored_users]):
+        if test_run:
+            remove_messages.append(_("{} will be removed from the delegates of {}.").format(user.full_name, represented_user.full_name))
+        else:
+            represented_user.delegates.remove(user)
+            remove_messages.append(_("Removed {} from the delegates of {}.").format(user.full_name, represented_user.full_name))
```

Comment:
as discussed in a comment above, this message and the one in line 320 are never shown because you're only using them in the bulk update when it's not a test run, correct? then i'd say you can remove the messages altogether.

Reply:
The only place where those messages are used is in line 782 in forms.py.
I am not sure whether we should care about different tenses or use one message tense everywhere.
Furthermore I placed this call into the user change form because the delegates and cc users should get removed on every user change. However, displaying the messages is handled individually outside the form (after changing a user)...

Reply:
I feel like this method is fine. Having a distinction between "will be changed" for a test run and "was changed" for a real run is a good thing imo. We should change the three calls in bulk_update_users below.

As far as I can see, in `bulk_update_users`, the simple idiomatic snippet:
```python
for user in deletable_users + users_to_mark_inactive:
    messages = remove_user_from_represented_and_ccing_users(user, deletable_users + users_to_mark_inactive, test_run)
    messages.warning(...)
```

should work -- or am I missing something? We may have to make the database transaction a bit bigger, but that shouldn't be much of a problem.